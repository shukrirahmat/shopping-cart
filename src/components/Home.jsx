import { Link } from "react-router-dom";
import photo from "../assets/images/storephoto.jpg"
import "../index.css"
import styles from "./Home.module.css";

function Home() {

    function handleScrollUp() {
        window.scrollTo({
            top: 0,
            behavior: "smooth"
        });
    }

    return (
        <div className={styles.wrapper}>
            <img src={photo} alt="Store Photo" className={styles.photo}/>
            <div className={styles.welcomeText}>
                <p class={styles.title}><b>Shops online with fast and reliable delivery</b></p>
                <p>Made from top-tier materials, check out of our high quality products! </p>
                <Link to="/shop" className={styles.shopBtn} onClick={handleScrollUp}>Shop now</Link>
            </div>
        </div>
    )
}

export default Home;