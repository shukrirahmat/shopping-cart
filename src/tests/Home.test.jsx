import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Home from "../components/Home.jsx";
import { BrowserRouter } from "react-router-dom";

describe("Home component", () => {
    it("renders home page", () => {
        const {container} = render(<BrowserRouter><Home/></BrowserRouter>);
        expect(container).toMatchSnapshot();
    })
})