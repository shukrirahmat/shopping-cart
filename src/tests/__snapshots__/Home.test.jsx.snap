// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders home page 1`] = `
<div>
  <div
    class="_wrapper_b140de"
  >
    <img
      alt="Store Photo"
      class="_photo_b140de"
      src="/src/assets/images/storephoto.jpg"
    />
    <div
      class="_welcomeText_b140de"
    >
      <p
        class="_title_b140de"
      >
        <b>
          Shops online with fast and reliable delivery
        </b>
      </p>
      <p>
        Made from top-tier materials, check out of our high quality products! 
      </p>
      <a
        class="_shopBtn_b140de"
        href="/shop"
      >
        Shop now
      </a>
    </div>
  </div>
</div>
`;
